### PR TITLE
add search functionality to plugin

### PR DESCRIPTION
**Pull request by richeduni**

### CODE REVIEW

File: admin/class-ed-solr-admin.php [hunk 1]
```diff
@@ -180,18 +180,9 @@ public function index_all_blogs_in_solr() {
 			$posts = get_posts( -1 );
 
 			foreach ( $posts as $post ) {
-				$document = $update->createDocument();
```
Comment (ajuszczyk) [inline admin/class-ed-solr-admin.php:183]: [x] Line 152: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 153: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 154: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 155: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 156: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 158: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 158: Opening statement of multi-line function call not indented correctly; expected 8 spaces but found 9
 [x] Line 159: Multi-line function call not indented correctly; expected 13 spaces but found 12
 [x] Line 168: Multi-line function call not indented correctly; expected 13 spaces but found 12
 [x] Line 169: Multi-line function call not indented correctly; expected 9 spaces but found 8
 [x] Line 183: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 184: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 195: Tabs must be used to indent lines; spaces are not allowed
 [ ] Line 257: The $domain arg must be a single string literal, not "$this->ed_solr".
 [ ] Line 258: The $domain arg must be a single string literal, not "$this->ed_solr".

File: includes/class-ed-solr-pagination.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * Define the internationalization functionality
```
Comment (ajuszczyk) [inline includes/class-ed-solr-pagination.php]: I think this description is not correct.

File: includes/class-ed-solr-pagination.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			'total'     => $search->get_total_pages(), // total amount of pages.
+			'current'   => $search->get_current_page(), // current page number.
+			'show_all'  => false, // set to true if you want to show all pages at once.
+			'mid_size'  => 2, // how much page numbers to show on the each side of the current page.
+			'end_size'  => 2, // how much page numbers to show at the beginning and at the end of the list.
+			'prev_next' => true, // if you set this to false, the previous and the next post links will be removed.
+			'prev_text' => '&laquo;', // «.
+			'next_text' => '&raquo;', // ».
+			'format'    => '',
+			'base'      => @add_query_arg( 'page', '%#%' ),
```
Comment (ajuszczyk) [inline includes/class-ed-solr-pagination.php]: Line 44: Silencing errors is discouraged

File: includes/class-ed-solr-post-mapper.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Fired during plugin activation
```
Comment (ajuszczyk) [inline includes/class-ed-solr-post-mapper.php]: Again, incorrect description.

File: includes/class-ed-solr-search.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Define the internationalization functionality
```
Comment (ajuszczyk) [inline includes/class-ed-solr-search.php]: to be removed.

File: includes/class-ed-solr-pagination.php (hunk 2 above)
Comment (richeduni) [reply includes/class-ed-solr-pagination.php]: updated

File: includes/class-ed-solr-post-mapper.php (hunk 4 above)
Comment (richeduni) [reply includes/class-ed-solr-post-mapper.php]: updated

File: includes/class-ed-solr-search.php (hunk 5 above)
Comment (richeduni) [reply includes/class-ed-solr-search.php]: updated

File: includes/class-ed-solr.php [hunk 6]
```diff
@@ -122,6 +122,22 @@ private function load_dependencies() {
 		 */
 		require_once plugin_dir_path( dirname( __FILE__ ) ) . 'public/class-ed-solr-public.php';
 
+		/**
```
Comment (ajuszczyk) [inline includes/class-ed-solr.php:111]: Another description comment to be removed from the top of the page

File: admin/class-ed-solr-admin.php (hunk 1 above)
Comment (richeduni) [reply admin/class-ed-solr-admin.php:183]: Hopefully the tabs have been fixed.

And 257 & 258 should work as they are run through __() translate function which return a literal.

File: public/class-ed-solr-public.php [hunk 7]
```diff
@@ -39,8 +39,8 @@ class Ed_Solr_Public {
 	 * Initialize the class and set its properties.
```
Comment (ajuszczyk) [inline public/class-ed-solr-public.php:30]: Another, remove description form the top.

File: tests/test-solr.php [hunk 8]
```diff
@@ -177,4 +177,88 @@ public function test_index_site() {
         $this->assertEquals(10, $resultSet->getNumFound());
         $this->assertEquals(0, $indexResult);
     }
+
```
Comment (ajuszczyk) [inline tests/test-solr.php:180]: Please run PHPCS on this and autofix/fix errors. Too many to put here.

File: public/class-ed-solr-public.php (hunk 7 above)
Comment (richeduni) [reply public/class-ed-solr-public.php:30]: No, think that is fine. Just describes what the constructor is doing

File: tests/test-solr.php (hunk 8 above)
Comment (richeduni) [reply tests/test-solr.php:180]: yeah, don't think it really matters if we meet the standards for tests.

Comment (ajuszczyk) [reply tests/test-solr.php:180]: ok

File: public/class-ed-solr-public.php (hunk 7 above)
Comment (ajuszczyk) [reply public/class-ed-solr-public.php:30]: I'm sorry Richard, I meant at line 1, not where I put the content in. It is difficult to do it on github.

File: admin/class-ed-solr-admin.php (hunk 1 above)
Comment (ajuszczyk) [reply admin/class-ed-solr-admin.php:183]: ok

File: includes/class-ed-solr-loader.php [hunk 9]
```diff
@@ -57,11 +57,11 @@ public function __construct() {
 	 * Add a new action to the collection to be registered with WordPress.
```
Comment (ajuszczyk) [inline includes/class-ed-solr-loader.php:46]: LINE 1 - remove extra description

File: admin/class-ed-solr-admin.php [hunk 10]
```diff
@@ -42,8 +42,8 @@ class Ed_Solr_Admin {
 	 * Initialize the class and set its properties.
```
Comment (ajuszczyk) [inline admin/class-ed-solr-admin.php:33]: LINE 1 - remove extra description

File: includes/class-ed-solr-search.php [hunk 11]
```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * Search solr for documents matching user search.
```
Comment (ajuszczyk) [inline includes/class-ed-solr-search.php]: Again, should be removed?

File: includes/class-ed-solr.php (hunk 6 above)
Comment (richeduni) [reply includes/class-ed-solr.php:111]: done

File: public/class-ed-solr-public.php (hunk 7 above)
Comment (richeduni) [reply public/class-ed-solr-public.php:30]: done

File: includes/class-ed-solr-loader.php (hunk 9 above)
Comment (richeduni) [reply includes/class-ed-solr-loader.php:46]: done

File: admin/class-ed-solr-admin.php (hunk 10 above)
Comment (richeduni) [reply admin/class-ed-solr-admin.php:33]: done

File: includes/class-ed-solr-search.php (hunk 11 above)
Comment (richeduni) [reply includes/class-ed-solr-search.php]: done

File: includes/class-ed-solr-search.php [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+	}
+
+	/**
+	 * Build up solr search query from search values
+	 *
+	 * @return string
+	 */
+	private function get_query_string() {
+        $query_string  = 'blogId:(' . implode( ' OR ', $this->blog_ids ) . ')';
```
Comment (ajuszczyk) [inline includes/class-ed-solr-search.php]: [x] Line 100: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 101: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 102: Tabs must be used to indent lines; spaces are not allowed
 [x] Line 104: Tabs must be used to indent lines; spaces are not allowed

Comment (richeduni) [reply includes/class-ed-solr-search.php]: done

**Pull request merged**